### DISCUSSION TITLE
independent line weights

### DISCUSSION THREAD

**Discussion by xarthurx** [libigl/libigl#1781] [Q&A]
For now, the line weight of lines is associated to the `ViewerData` class, meaning one `ViewerData` can only have one "line weights" for all the lines.

I currently have around 10k lines and would like to draw them with different line weights...

Am I supposed to create a `ViewerData` for each of the line and put them separately? Or is there a better way to do this?
It seems the by design, creating a `ViewerData` for each line should not be the correct way...

**Comment by alecjacobson** [comment]
do you mean widths? I think on macos the widths don't even work :-(

**Reply by xarthurx** [reply]
Yes, width. I'm on Windows.
Then, if drawing lines as OpenGL cylinder, is it possible to easily add some function on the user side, so that the viewer can understand?

**Reply by jdumas** [reply]
I have an old PR on that topic (#875). It was never merged, but I believe that would offer a better solution to support variable line width.

**Reply by xarthurx** [reply]
I'll check.
For the cylinder approach, I found that there is a `igl::cylinder`. But I would actually mean a `cone cylinder` shape where I can define different `R` of the two end.

**Reply by xarthurx** [reply]
This question is a bit of a special need actually.
I actually would like to draw ~10k lines, each has a different `R` on the two ends, and interpolate linearly in-between.

**Reply by jdumas** [reply]
The code in this PR should be able to handle that (that's how I draw arrows). You might need to update it though, as the PR is pretty old.

**Reply by xarthurx** [reply]
@jdumas  I tried to merge that PR by cherry-picking the commits related to the files in that PR. But a lot of things has changed and many OpenGL-related settings. 

The complication works, but the line now doesn't show up anymore. 
I guess this is sth beyond my ability at the moment (no familiar with OpenGL).

May I ask why that PR is abandoned? Seems quite useful...

**Reply by jdumas** [reply]
I never had time to finish integrating it properly.
